ActionMenu/ActionMenu.cs

```diff
@@ -586,28 +588,37 @@ public static Menus SplitOverCrowdedMenus(Menus m, uint overcrowdedMenuValue = 7
                 case SettingsType.GameObjectDropdown: {
                     var submenuName = Path(menuPrefix, s.name);
                     // if parameter name has suffix Impulse, adapt control type
-                    var isImpulse = s.machineName.EndsWith("Impulse");
+                    var (parentControl, parentDuration) = DetectImpulseToggle(s);
                     var dd = s.dropDownSettings;
                     var selectedValue = animator?.GetInteger(s.machineName) ?? dd?.defaultValue;
 
                     item.action = new ItemAction
                     {
                         type = "menu",
-                        menu = submenuName,
+                        menu = StripMenuNameTags(submenuName),
                     };
 
-                    List<MenuItem> sitems = dd.options.Select((o, index) => new MenuItem
-                    {
-                        name = o.name,
-                        enabled = index == selectedValue,
-                        action = new ItemAction
-                        {
-                            type = "avatar parameter",
-                            parameter = s.machineName,
-                            control = isImpulse ? "impulse" : "toggle",
-                            value = index,
-                            exclusive_option = !isImpulse,
-                        },
+                    List<MenuItem> sitems = dd.options.Select((o, index) => {
+                        var (control, duration) = DetectImpulseToggle(s);

```

Comment:
This will do the same as line 591 above:
>                     var (parentControl, parentDuration) = DetectImpulseToggle(s);

here you probably meant to pass `o` but it's different type so the function need to be more generic. This would mean the parent can decide the type and duration if the child didn't specify it (override). That behaviour should be ok.